DataFormats/ParticleFlowCandidate/interface/PFCandidate.h

```diff
@@ -386,29 +387,7 @@ namespace reco {
 
     friend std::ostream& operator<<(std::ostream& out, const PFCandidate& c);
 
-    //Tips on setting the vertex efficiently
-    //There are two choices: a) use the vertex_ data member, or b) point to the vertex
-    //of one of the refs stored by this class. The PFVertexType enum gives the current list
-    //of possible references. For these references, use the setVeretxSource method and NOT
-    //the setVertex method. If none of the available refs have the vertex that you want for this
-    //PFCandidate, use the setVertex method. If you find that you are using frequently two store a
-    // vertex that is the same as one of the refs in this class, you should just extend the enum
-    // and modify the vertex() method accordingly.
-    void setVertexSource(PFVertexType vt) {
-      vertexType_ = vt;
-      if (vertexType_ != kCandVertex)
-        LeafCandidate::setVertex(Point(0., 0., 0.));
-    }
-
-    void setVertex(const math::XYZPoint& p) override {
-      LeafCandidate::setVertex(p);
-      vertexType_ = kCandVertex;
-    }
-
     const Point& vertex() const override;
-    double vx() const override { return vertex().x(); }
-    double vy() const override { return vertex().y(); }
-    double vz() const override { return vertex().z(); }
```

Comment:
shouldn't these be kept?
otherwise, IIUC, the base class is going to return a wrong result compared to `vertex().x()`
https://github.com/cms-sw/cmssw/blob/a7d7e5843820f0bca99289a87931245338e78ffb/DataFormats/Candidate/interface/LeafCandidate.h#L167

Reply:
indeed - thanks!

Reply:
...for the cases when reading legacy files and calling the vx/vy/vz methods directly, which may not happen in practice. it would be an odd bug nevertheless, so let's put them back